module-2/backend/week_two.md

```diff
@@ -26,21 +37,48 @@ end
 
 Now how would you find the owner of the team with an id of 4?
 
+team = Team.find(4)
+team.owner
+
 3. What do they allow you to do?
+I'm confused. Who are they?
 7. In a database that's holding students and teachers, what will be the relationship between students and teachers? Draw the schema diagram.
+
+teacher (n) |has| (n) student
+
 8. Define foreign key, primary key, and schema.
+Primary key is the unique identifier for a row. 
+Foreign key is the primary key of a row in another table that is linked to the current table.
+Schema is the overall database design.
+
 9. Describe the relationship between a foreign key on one table and a primary key on another table.
+The primary key of a table is the foreign key on the table it is linked to. Ex: Student - student_id (primary key), name, teacher_id
+
 10. What are the parts of an HTTP response?
+I don't really understand. Do you mean params? 
+
 11. Describe some techniques to make our Sinatra code more DRY. Give an example of when you would use these techniques.
+using a partial for html forms, so we don't have to repeat it.
 
 
 ### Optional Questions
 
 1. Name your five favorite ActiveRecord methods (i.e. methods your models inherit from ActiveRecord) and describe what they do.
+group - does the equivalent of group_by in SQL. it's an aggregate function.
+select - selects a given set of columns from the tables
+find_by - equivalent to WHERE 
```

Comment:
Assume because of the capitalization that you're referring to a SQL WHERE statement. Be careful here. ActiveRecord also has a `.where`, and it functions slightly differently from `.find_by`.

`.where` will return multiple records, while `.find_by` returns only one.